app/services/teacher_training_public_api/sync_all_providers_and_courses.rb

```diff
@@ -1,17 +1,16 @@
 module TeacherTrainingPublicAPI
   class SyncAllProvidersAndCourses
     def self.call
-      begin
-        (1..).each do |page|
-          sync_providers(
-            TeacherTrainingPublicAPI::Provider
-              .where(year: RecruitmentCycle.current_year)
-              .paginate(page: page, per_page: 500)
-              .all,
-          )
-        end
-      rescue JsonApiClient::Errors::ClientError
-        # This is how the API responds when we run out of pages :/
+      is_last_page = false
+      page_number = 0
+      until is_last_page
+        page_number += 1
+        response = TeacherTrainingPublicAPI::Provider
+          .where(year: RecruitmentCycle.current_year)
+          .paginate(page: page_number, per_page: 500)
+          .all
+        sync_providers(response)
+        is_last_page = true if response.links.links['next'].nil?
       end
```

Comment:
So from the docs i can see that if you can actually call response.pages.next to return the next page of results. Weirdly though, it's not retaining the per_page value. I'm wondering if it's not being passed into the links on the TTAPI. If it were working correctly we could do.

```
response = TeacherTrainingPublicAPI::Provider
         .where(year: RecruitmentCycle.current_year)
         .paginate(page: page_number, per_page: 500)
         .all

loop do
  sync_providers(response)
  response = pages.next
  break if response.links.links['next'].nil?
end 
```

I'm not sure it's worth holding this up for that though considering your implementation is working nicely and we would have to deploy out the TTAPI etc.

Reply:
Interesting, let's pick it up on Monday 👍

Reply:
Sounds good. I'm just going to double-check if pagination is working correctly on the TTAPI, mainly because Sau implemented it and i can get some sweet revenge on him for rejecting every PR i ever opened on the Find team haha